Microsoft.Toolkit.Uwp.UI.Controls/RadialProgressBar/RadialProgressBar.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            base.OnMinimumChanged(oldMinimum, newMinimum);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Called when the Maximum property changes.
+        /// </summary>
+        /// <param name="oldMaximum">Old value of the Maximum property.</param>
+        /// <param name="newMaximum">New value of the Maximum property.</param>
+        protected override void OnMaximumChanged(double oldMaximum, double newMaximum)
+        {
+            base.OnMaximumChanged(oldMaximum, newMaximum);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Called when the Value property changes.
+        /// </summary>
+        /// <param name="oldValue">Old value of the Value property.</param>
+        /// <param name="newValue">New value of the Value property.</param>
+        protected override void OnValueChanged(double oldValue, double newValue)
+        {
+            base.OnValueChanged(oldValue, newValue);
+            RenderSegment();
+        }
+
+        /// <summary>
+        /// Update the visual state of the control when its template is changed.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            outlineFigure = GetTemplateChild(OutlineFigurePartName) as PathFigure;
+            outlineArc = GetTemplateChild(OutlineArcPartName) as ArcSegment;
+            barFigure = GetTemplateChild(BarFigurePartName) as PathFigure;
+            barArc = GetTemplateChild(BarArcPartName) as ArcSegment;
+
+            allTemplatePartsDefined = outlineFigure != null && outlineArc != null && barFigure != null && barArc != null;
+
+            RenderAll();
+        }
+
+        /// <summary>
+        /// Gets or sets the thickness of the circular ouline and segment
+        /// </summary>
+        public double Thickness
+        {
+            get { return (double)GetValue(ThicknessProperty); }
+            set { SetValue(ThicknessProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the Thickness dependency property
+        /// </summary>
+        public static readonly DependencyProperty ThicknessProperty = DependencyProperty.Register(nameof(Thickness), typeof(double), typeof(RadialProgressBar), new PropertyMetadata(0.0, ThicknessChangedHandler));
+
+        /// <summary>
+        /// Gets or sets the color of the circular ouline on which the segment is drawn
+        /// </summary>
+        public Brush Outline
+        {
+            get { return (Brush)GetValue(OutlineProperty); }
+            set { SetValue(OutlineProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the Outline dependency property
+        /// </summary>
+        public static readonly DependencyProperty OutlineProperty = DependencyProperty.Register(nameof(Outline), typeof(Brush), typeof(RadialProgressBar), new PropertyMetadata(new SolidColorBrush(Windows.UI.Colors.Transparent)));
```

Comment:
Would it make sense to use the same property names as ProgressBar for the Background color? In other words, wondering if this property is needed?